key.ui/src/main/java/org/key_project/util/java/CheckedProcessBuilder.java

```diff
@@ -0,0 +1,74 @@
+/* This file is part of KeY - https://key-project.org
+ * KeY is licensed under the GNU General Public License Version 2
+ * SPDX-License-Identifier: GPL-2.0-only */
+package org.key_project.util.java;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Process builder that first checks whether a program is installed.
+ *
+ * @author Arne Keller
+ */
+public class CheckedProcessBuilder {
+    public static final Logger LOGGER = LoggerFactory.getLogger(CheckedProcessBuilder.class);
+
+    /**
+     * The programs already checked.
+     */
+    private static final Set<String> CHECKED = new HashSet<>();
+    /**
+     * Installed programs.
+     */
+    private static final Set<String> AVAILABLE = new HashSet<>();
+    /**
+     * Program of this builder.
+     */
+    private final String program;
+
+    /**
+     * Create a new builder and check for the given program.
+     *
+     * @param program the program
+     * @param programCheck program invocation used to check (e.g. z3 --version)
+     */
+    public CheckedProcessBuilder(String program, String[] programCheck) {
```

Comment:
The interface seems a bit strange here, since you have to pass the name of the executable three times:
<pre>
new CheckedProcessBuilder(<b>exe</b>, new String[] { <b>exe</b>, "-?" }).start(<b>exe</b>, "-a", "KeY", title, text);
</pre>
I think this can be simplified.